packages/sessions-sdk-react/src/components/session-provider.tsx

```diff
@@ -719,6 +712,14 @@ const useSessionState = ({
     ],
   );
 
+  /** refs */
+  const connectWalletRef = useRef(connectWallet);
+
+  /** effects */
+  useEffect(() => {
+    connectWalletRef.current = connectWallet;
+  });
```

Comment:
whoa what is going on here?  This seems like a hack around another issue and not a proper fix

Reply:
Yes, this is correct

there were a number of issues here, with the biggest offender being that the connect wallet callback was being used in another callback, but wasn’t set as a dep in that callback’s dep array, which could result in runtime bugs where the function is old and points to incorrect variables.

simply adding it as a dep array entry blew up the build, because it meant that connect wallet was being used before it was defined.

I made this latest Ref change to work around this, while ensuring connect wallet is always updated

Reply:
Yeah I really don't like this fix, I think we should either leave it as it is and add lint ignores or fix it properly.  This fix doesn't actually resolve the underlying issue and makes it even more hacky.